### PR TITLE
Add section about development best practices to CONTRIBUTING

### PR DESCRIPTION
**Pull request by kestarumper**
This PR adds a section about development best practices and how to achieve better DX when working with this codebase.

### CODE REVIEW

**Review by piotrpospiech — APPROVED** [review]

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -30,6 +30,36 @@ Our kanban board is public and available [here](https://github.com/orgs/vazco/pr
 - Make sure you have added the necessary tests for your changes. Do not worry though, the Codecov bot will report it in the pull request.
 - Make sure your code passes _all_ tests: `npm test`.
 
+### Development
+
+For the best developer experience (DX) when developing locally it is recommended to start `build` and `test` in a `--watch` mode.
```
Comment (wadamek65) [inline .github/CONTRIBUTING.md:35]: The `a` is redundant in "a --watch mode". Should either be just "in --watch mode" or "in the --watch mode".

In general you could simplify the start of this sentence but it's subjective so I'm okay with this version too. "For the best local developer experience"

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -30,6 +30,36 @@ Our kanban board is public and available [here](https://github.com/orgs/vazco/pr
...
+
+In the root directory...
```
Comment (wadamek65) [inline .github/CONTRIBUTING.md:37]: This sounds a little strange. Maybe reword this to a direct command like "Run the commands below in the root directory."

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -30,6 +30,36 @@ Our kanban board is public and available [here](https://github.com/orgs/vazco/pr
...
+
+The local version of docs will use the locally built version of uniforms, so make sure to run the build in watch mode for live changes.
+
+Navigate to `/website` and run
+
+```sh
+npm start
+```
+
+it will start the docusaurus in development mode supporting hot reload, so you should see the changes made to the website immediately.
```
Comment (wadamek65) [inline .github/CONTRIBUTING.md:61]: I think this sentence should be capitalized. Also not sure if you need the comma after "hot reload"

**Review by wadamek65 — APPROVED** [review]

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -30,6 +30,36 @@ Our kanban board is public and available [here](https://github.com/orgs/vazco/pr
...
+```sh
+npm run build -- --watch
+```
+
+#### Running the tests in watch mode
+
+```sh
+npm run test -- --watch
+```
+
```
Comment (Monteth) [inline .github/CONTRIBUTING.md:50]:
```suggestion
#### Fixing lint issues automatically

```sh
npm run lint:code -- --fix

```
```

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (kestarumper) [reply .github/CONTRIBUTING.md:35]: That's better!
